evaluation/swe_bench/eval_infer.py

```diff
@@ -94,13 +95,28 @@ def get_config(instance: pd.Series) -> AppConfig:
 
 def process_instance(
     instance: pd.Series,
-    metadata: EvalMetadata | None = None,
+    _metadata: EvalMetadata,
```

Comment:
Is this change is necessary or not?

Also, we could probably feed this `metadata` to `EvalOutput` now?

Reply:
> Is this change is necessary or not?

The parameter isn't used at the moment, so this isn't necessary for the script to work. A few reasons why I made the change:

1. We should always have a metadata object, so there's no reason to pass `None`.
2. `run_evaluation` expects a function of type `Callable[[pd.Series, EvalMetadata, bool], EvalData]`. If `run_evaluation` is following its type guarantees, this function will never be called with `None`. 

> Also, we could probably feed this `metadata` to `EvalOutput` now?

I hadn't seen that was a field on `EvalOutput`. I'll pass it through.

Reply:
But I don't think point 2 is true. It'd be good to work towards type correctness when possible but happy to revert if you prefer.

Reply:
Sounds good!